server/http4s-server/src/main/scala/sttp/tapir/server/http4s/EndpointToHttp4sServer.scala

```diff
@@ -1,70 +1,75 @@
 package sttp.tapir.server.http4s
 
+import cats.~>
 import cats.data._
 import cats.effect.{ContextShift, Sync}
 import cats.implicits._
-import org.http4s.{EntityBody, HttpRoutes, Request, Response}
+import org.http4s.{EntityBody, HttpRoutes, Http, Request, Response}
 import org.log4s._
 import sttp.tapir.monad.MonadError
 import sttp.tapir.server.internal.{DecodeInputsResult, InputValues, InputValuesResult}
 import sttp.tapir.server.{DecodeFailureContext, DecodeFailureHandling, ServerDefaults, ServerEndpoint, internal}
 import sttp.tapir.{DecodeResult, Endpoint, EndpointIO, EndpointInput}
+import cats.arrow.FunctionK
 
 class EndpointToHttp4sServer[F[_]: Sync: ContextShift](serverOptions: Http4sServerOptions[F]) {
   private val outputToResponse = new OutputToHttp4sResponse[F](serverOptions)
 
-  def toRoutes[I, E, O](se: ServerEndpoint[I, E, O, EntityBody[F], F]): HttpRoutes[F] = {
-    val service: HttpRoutes[F] = HttpRoutes[F] { req: Request[F] =>
-      def decodeBody(result: DecodeInputsResult): F[DecodeInputsResult] = {
-        result match {
-          case values: DecodeInputsResult.Values =>
-            values.bodyInput match {
-              case Some(bodyInput @ EndpointIO.Body(bodyType, codec, _)) =>
-                new Http4sRequestToRawBody(serverOptions).apply(req.body, bodyType, req.charset, req).map { v =>
-                  codec.decode(v) match {
-                    case DecodeResult.Value(bodyV)     => values.setBodyInputValue(bodyV)
-                    case failure: DecodeResult.Failure => DecodeInputsResult.Failure(bodyInput, failure): DecodeInputsResult
-                  }
+  def toHttp[I, E, O, G[_]: Sync](se: ServerEndpoint[I, E, O, EntityBody[F], G])(t: F ~> G): Http[OptionT[G, *], F] = {
```

Comment:
`t` should probably go first here (and in other `toHttp` function) as well, to match the syntax.